store/CSDK/main.cpp

```diff
@@ -0,0 +1,239 @@
+            jobject element = env->GetObjectArrayElement(row, j);
+            char *str = (char *) env->GetStringUTFChars((jstring) element, JNI_FALSE);
+            printf("%s\t", str);
+        }
+        printf("\n");
+    }
+
+    carbonReaderClass.close();
+}
+
+
+/**
+ * read data from S3
+ * parameter is ak sk endpoint
+ *
+ * @param env jni env
+ * @param argv argument vector
+ * @return
+ */
+bool readFromS3_2(JNIEnv *env, char *argv[]) {
+    CarbonReader carbonReaderClass;
+
+    char *args[3];
+    // "your access key"
+    args[0] = argv[1];
+    // "your secret key"
+    args[1] = argv[2];
+    // "your endPoint"
+    args[2] = argv[3];
+
+    carbonReaderClass.builder(env, "s3a://sdk/WriterOutput", "test");
+    carbonReaderClass.build(args[0], args[1], args[2]);
+
+    printf("\nRead data from S3:\n");
+    while (carbonReaderClass.hasNext()) {
+        jobjectArray row = carbonReaderClass.readNextRow();
+        jsize length = env->GetArrayLength(row);
+
+        int j = 0;
+        for (j = 0; j < length; j++) {
+            jobject element = env->GetObjectArrayElement(row, j);
+            char *str = (char *) env->GetStringUTFChars((jstring) element, JNI_FALSE);
+            printf("%s\t", str);
+        }
+        printf("\n");
+    }
+
+    carbonReaderClass.close();
+}
+
+/**
+ * read data from S3
+ * parameter is ak sk endpoint
+ *
+ * @param env jni env
+ * @param argv argument vector
+ * @return
+ */
+bool readFromS3(JNIEnv *env, char *argv[]) {
+    CarbonReader carbonReaderClass;
+
+    char *args[3];
+    // "your access key"
+    args[0] = argv[1];
+    // "your secret key"
+    args[1] = argv[2];
+    // "your endPoint"
+    args[2] = argv[3];
+
+    carbonReaderClass.builder(env, "s3a://sdk/WriterOutput", "test");
+    carbonReaderClass.build(3, args);
+
+    printf("\nRead data from S3:\n");
+    while (carbonReaderClass.hasNext()) {
+        jobjectArray row = carbonReaderClass.readNextRow();
+        jsize length = env->GetArrayLength(row);
+
+        int j = 0;
+        for (j = 0; j < length; j++) {
+            jobject element = env->GetObjectArrayElement(row, j);
+            char *str = (char *) env->GetStringUTFChars((jstring) element, JNI_FALSE);
+            printf("%s\t", str);
+        }
+        printf("\n");
+    }
+
+    carbonReaderClass.close();
+}
+
+
+/**
+ * This a example for C++ interface to read carbon file
+ * If you want to test read data fromS3, please input the parameter: ak sk endpoint
+ *
+ * @param argc argument counter
+ * @param argv argument vector
+ * @return
+ */
+int main(int argc, char *argv[]) {
+    // init jvm
+    JNIEnv *env;
+    env = initJVM();
+
+    if (argc > 3) {
+        readFromS3(env, argv);
+    } else {
+        readFromLocal(env);
+        readFromLocalWithoutProjection(env);
+    }
+    cout << "destory jvm\n\n";
+    (jvm)->DestroyJavaVM();
+
+    cout << "\nfinish destory jvm";
+    fprintf(stdout, "Java VM destory.\n");
+    return 0;
+}
+
```

Comment:
There are some binary files added in this PR, please remove them

Reply:
ok, done